craft_providers/multipass/multipass_instance.py

```diff
@@ -0,0 +1,395 @@
+
+            uid_map = {host_uid: "0"}
+            gid_map = {host_gid: "0"}
+
+        self._multipass.mount(
+            source=host_source,
+            target=f"{self.name}:{target.as_posix()}",
+            uid_map=uid_map,
+            gid_map=gid_map,
+        )
+
+    def pull_file(self, *, source: pathlib.Path, destination: pathlib.Path) -> None:
+        """Copy a file from the environment to host.
+
+        :param source: Environment file to copy.
+        :param destination: Host file path to copy to.  Parent directory
+            (destination.parent) must exist.
+
+        :raises FileNotFoundError: If source file or destination's parent
+            directory does not exist.
+        :raises MultipassError: On unexpected error copying file.
+        """
+        proc = self.execute_run(["test", "-f", source.as_posix()], check=False)
+        if proc.returncode != 0:
+            raise FileNotFoundError(f"File not found: {source.as_posix()!r}")
+
+        if not destination.parent.is_dir():
+            raise FileNotFoundError(f"Directory not found: {str(destination.parent)!r}")
+
+        self._multipass.transfer(
+            source=f"{self.name}:{source.as_posix()}", destination=str(destination)
+        )
+
+    def push_file(self, *, source: pathlib.Path, destination: pathlib.Path) -> None:
+        """Copy a file from the host into the environment.
+
+        :param source: Host file to copy.
+        :param destination: Target environment file path to copy to.  Parent
+            directory (destination.parent) must exist.
+
+        :raises FileNotFoundError: If source file or destination's parent
+            directory does not exist.
+        :raises MultipassError: On unexpected error copying file.
+        """
+        if not source.is_file():
+            raise FileNotFoundError(f"File not found: {str(source)!r}")
+
+        proc = self.execute_run(
+            ["test", "-d", destination.parent.as_posix()], check=False
+        )
+        if proc.returncode != 0:
+            raise FileNotFoundError(
+                f"Directory not found: {str(destination.parent.as_posix())!r}"
+            )
+
+        self._multipass.transfer(
+            source=str(source),
+            destination=f"{self.name}:{destination.as_posix()}",
+        )
+
+    def start(self) -> None:
+        """Start instance.
+
+        :raises MultipassError: On unexpected failure.
+        """
+        self._multipass.start(instance_name=self.name)
+
+    def stop(self, *, delay_mins: int = 0) -> None:
+        """Stop instance.
+
+        :param delay_mins: Delay shutdown for specified minutes.
+
+        :raises MultipassError: On unexpected failure.
+        """
+        self._multipass.stop(instance_name=self.name, delay_mins=delay_mins)
+
+    def supports_mount(self) -> bool:  # pylint: disable=no-self-use
+        """Check if instance supports mounting from host.
+
+        :returns: True if mount is supported.
```

Comment:
The "if" here is a leftover from copying the docstring, right? Because this always returns True...

Reply:
Correct. I originally wanted the interface in place should Multipass someday not support mounting in a particular case (like is the case for LXD).  However, after reconsidering it, maybe it is best we just drop the interface entirely until it's a requirement.